src/bin/am/commands/start.rs

```diff
@@ -83,10 +84,110 @@ pub struct Arguments {
+                .map(|url| {
+                    let num = COUNTER.fetch_add(1, Ordering::SeqCst);
+                    InnerEndpoint::new(url, format!("am_{num}"))
+                })
+                .collect()
+        } else if let Some(endpoints) = config.endpoints {
+            endpoints
+                .into_iter()
+                .map(|endpoint| {
+                    let job_name = endpoint.job_name.unwrap_or_else(|| {
+                        format!("am_{num}", num = COUNTER.fetch_add(1, Ordering::SeqCst))
+                    });
+                    InnerEndpoint::new(endpoint.url, job_name)
+                })
+                .collect()
+        } else {
+            Vec::new()
+        };
+
+        InnerArguments {
+            metrics_endpoints,
+            prometheus_version: args.prometheus_version,
+            listen_address: args.listen_address,
+            pushgateway_enabled: args
+                .pushgateway_enabled
+                .or(config.pushgateway_enabled)
+                .unwrap_or(false),
+            pushgateway_version: args.pushgateway_version,
+            ephemeral_working_directory: args.ephemeral,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct InnerEndpoint {
+    pub url: url::Url,
+    pub job_name: String,
+}
+
+impl InnerEndpoint {
+    fn new(url: url::Url, job_name: String) -> Self {
+        Self { url, job_name }
+    }
+}
+
+impl From<InnerEndpoint> for prometheus::ScrapeConfig {
+    /// Convert an InnerEndpoint to a Prometheus ScrapeConfig.
+    ///
+    /// Scrape config only supports http and https atm.
+    fn from(endpoint: InnerEndpoint) -> Self {
+        let scheme = match endpoint.url.scheme() {
+            "http" => Some(prometheus::Scheme::Http),
+            "https" => Some(prometheus::Scheme::Https),
+            _ => None,
+        };
+
+        let mut metrics_path = endpoint.url.path();
+        if metrics_path.is_empty() {
+            metrics_path = "/metrics";
+        }
+
+        let host = match endpoint.url.port() {
+            Some(port) => format!("{}:{}", endpoint.url.host_str().unwrap(), port),
+            None => endpoint.url.host_str().unwrap().to_string(),
+        };
+
+        prometheus::ScrapeConfig {
+            job_name: endpoint.job_name,
+            static_configs: vec![prometheus::StaticScrapeConfig {
+                targets: vec![host],
+            }],
+            metrics_path: Some(metrics_path.to_string()),
+            scheme,
+        }
+    }
+}
+
+pub async fn handle_command(args: Arguments, config: AmConfig, mp: MultiProgress) -> Result<()> {
+    let mut args = InnerArguments::new(args, config);
+
+    if args.metrics_endpoints.is_empty() && args.pushgateway_enabled {
+        // info!("No metrics endpoints provided and pushgateway is not enabled. Please provide an endpoint.");
+        // let endpoint = interactive::user_input("Endpoint")?;
+        // args.metrics_endpoints.push(endpoint_parser(&endpoint)?);
```

Comment:
why comemnted out

Reply:
The comments have been removed and the code is refactored.